apps/cli/README.md

```diff
@@ -0,0 +1,281 @@
+```
+
+> [!TIP]
+> When users attempt to approve an upgrade with their wallets, the only data displayed is the upgrade ID. To ensure
+> they're approving the correct upgrade, users must recalculate the ID locally and verify that it matches the one shown in
+> their wallets. This is why id is a standalone command in the CLI tool.
+
+#### **`check -f <rawUpgradeDir> --ref=<commitRef>`**
+
+This command checks that the upgrade can be executed. Then, it summarizes the changes produced by the upgrade. The
+output has several sections:
+
+- **Metadata:** General information about the upgrade
+- **Facet changes**: Summary of facet changes for the main ZKsync diamond.
+- **Property changes:** Shows changes in the most important properties of the main diamond.
+- **System contract changes:** Lists system contracts being modified by the upgrade.
+
+```bash
+pnpm validate check -f <rawUpgradeDir> --ref=<commitRef>
+```
+
+```bash
+pnpm validate check -f test/data/sample-upgrade.json --ref=main
+
+✔ Locally compiling system contracts
+✔ Gathering current zksync state
+✔ Simulating upgrade
+✔ Generating report
+
+[...]
+```
+
+#### **`storage-diff -f <rawUpgradeDir> --ref=<commitRef>`**
+
+This command simulates the upgrade and then shows how the storage of the main diamond gets modified by the upgrade.
+
+```bash
+pnpm validate storage-diff -f <rawUpgradeDir> --ref=<commitRef>
+```
+
+```bash
+pnpm validate storage-diff -f test/data/sample-upgrade.json --ref=main
+
+✔ Gathering current zksync state
+✔ Forking network
+✔ Simulating upgrade
+✔ Gathering final state
+✔ Calculating report
+
+[...]
+```
+
+#### **`facet-diff <upgradeDir> <facetName>`**
+
+Shows the proposed changes in a specified facet.
+
+```bash
+pnpm validate show-diff <upgradeDir> <facetName>
+```
+
+```bash
+pnpm validate facet-diff ../zksync-era/etc/upgrades/1699353977-boojum GettersFacet --ref=e77971dba8f589b625e72e69dd7e33ccbe697cc0 #Example with GettersFacet in Boojum upgrade with specific commit reference.
+```
+
+#### **`verifier-diff <upgradeDir>`**
+
+Shows the proposed changes between current verifier source code and the proposed one.
+
+```bash
+pnpm validate verifier-diff <upgradeDir>
+```
+
+```bash
+pnpm validate verifier-diff ../zksync-era/etc/upgrades/1699353977-boojum #Example of verifier-diff command for Boojum upgrade.
+```
+
+#### **`download-diff <upgradeDir> <targetSourceCodeDir>`**
+
+Downloads both the current and proposed versions of each contract being upgraded for comparison.
+
+1. **Run the Command:**
+
+   ```bash
+     pnpm validate download-diff <upgradeDir> <targetSourceCodeDir>
+   ```
+
+   `<targetSourceCodeDir>`: The directory where you wish to save the downloaded differences.
+
+   _Note: Depending on the specific upgrade referenced, the `--ref` option might be necessary. For more info, please
+   refer to [--ref in options section.](#🎛️-options)_
```

Comment:
This link doesn't work. Removing the emoji fix it. Maybe there is a way to keep the emoji and make the link work. But I wasn't able to make it work.